src/corelibs/U2Core/src/globals/Task.h

```diff
@@ -589,8 +589,16 @@ class U2CORE_EXPORT TaskScheduler : public QObject {
 
 signals:
     void si_noTasksInScheduler();
+
+    // TODO: remove this signal from TaskScheduler. It is not scheduler responsibliity to track global UGENE state.
```

Comment:
I suppose, it's much better just to rename this signal, cuz it's pretty useful, but has a strange name. It's emitted when the all top level tasks and their subtasks are finished their jobs, so you may name it something like `si_allTopLevelTasksAreDone`

Reply:
The fix is out of scope of this PR. I have only documented the lame code and will address it separately